### PR TITLE
Add proper Dockerfile

### LINKED ISSUE #111: Docker support
Is it possible to run this in one (or more) Docker containers? I'd really like to know.

I suppose I could try to write a docker-compose file but my knowledge of Docker, networking, email standards, etc are not up to the task. So I'd feel more comfortable if someone more knowledgeable could pick up this task.

### PR DESCRIPTION
**Pull request by dapetcu21**
The Dockerfile that @UnoSD posted had a number of issues (like using an ancient version of node and reading the dkim key from the wrong path). I cleaned it up.

Additionally, I added some more configurability in `index.js` via environment variables. Currently, most of the stuff in the init section at the end was hardcoded for the `forwardemail.net` domain. I tried to keep it backwards compatible, so as not to break your deployment, but I honestly think a few things, like the paths to the keys should live outside of the repo, in the pm2 environment, as they are deployment-specific.

I also added an IP_ADDRESS environment variable to make this work from behind a NAT.

Stuff that the Dockerfile still can't do properly:
* [ ] SpamAssassin calls time out (I'll dig a bit further into this)
* [x] SSL (shouldn't be hard to add, if anyone needs it)

Fixes #111

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1110,25 +1112,34 @@ class ForwardEmail {
...
-    // needsUpgrade = true;
+
+  if (process.env.SECURE === 'true') {
     config.ssl = {
-      secure: process.env.SECURE === 'true',
-      key: fs.readFileSync('/home/deploy/mx1.forwardemail.net.key', 'utf8'),
-      cert: fs.readFileSync('/home/deploy/mx1.forwardemail.net.cert', 'utf8'),
-      ca: fs.readFileSync('/home/deploy/mx1.forwardemail.net.ca', 'utf8')
+      secure: true,
+      key: fs.readFileSync(process.env.SSL_KEY || '/home/deploy/mx1.forwardemail.net.key', 'utf8'),
```
Comment (tracker1) [inline index.js]: Shouldn't this be...
```
key: process.env.SSL_KEY || fs.readFileSync('/home/deploy/mx1.forwardemail.net.key', 'utf8'),
```

Comment (dapetcu21) [reply index.js]: I guess it depends wether you want to provide it in an env var directly or read it from a file. I think both options are valid. Maybe I should add both. (SSL_KEY and SSK_KEY_FILE)

Comment (dapetcu21) [reply index.js]: And that way, it will also be easier to add the keys from Docker. You just specify them as env vars when running `docker run`.

Comment (dapetcu21) [reply index.js]: Fixed

**Comment by niftylettuce** [comment]
I have updated the configuration to support a dotenv configuration.

See `.env.schema`, `.env.defaults`, and you can add a `.env` of your own with overrides.  My local `.env` file has just one property needed of `DKIM_PRIVATE_KEY_PATH`, however you will need others to override defaults specified in `.env.defaults`.  This makes using this package much easier for you I would think.  There is also a new `app.js and `proxy.js` file which should help you with graceful reloading, pm2 deployment, and certbot.  See the updated `ecosystem.json` file as well.  If you wish to submit a new PR with a Dockerfile considering all this, I would gladly accept immediately.

The `mx1.forwardemail.net` and `mx2.forwardemail.net` servers contain a `.env` file that looks like this, if it is helpful as a resource for you for production:

```txt
DKIM_PRIVATE_KEY_PATH=/home/deploy/.dkim.key
SMTP_SSL_KEY_PATH=/home/deploy/certbot/config/live/mx1.forwardemail.net/privkey.pem
SMTP_SSL_CERT_PATH=/home/deploy/certbot/config/live/mx1.forwardemail.net/fullchain.pem
SMTP_EXCHANGE_DOMAINS=mx1.forwardemail.net,mx2.forwardemail.net
CERTBOT_WELL_KNOWN_NAME=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
CERTBOT_WELL_KNOWN_CONTENTS=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
```

**Pull request closed without merging**